### PR TITLE
CH8527 | Update The Pennylane Demos Landing Page 

### PR DESCRIPTION
**Pull request by AndrewGardhouse**
[CH8527](https://app.clubhouse.io/xanaduai/story/8527/update-the-pennylane-demos-landing-page)

Screen Shot 2021-09-13 at 4 30 00 PM

### CODE REVIEW

**Comment by josh146** [comment]
@AndrewGardhouse looking really nice so far! I only have two main comments:

- The main demos page, the cards are all different heights. This is partially my fault - when I first made it, I wasn't sure how to get a 'grid' pattern, and so re-sized all the images so that the vertical card spacing was identical. But it would be nice to enforce this in the CSS, rather than by image + text height!

- The 'back' button on each category page _and_ the demos themselves takes you to the main QML page, rather than the demos page:
  
  image

  This should just be a matter of changing the location of the Sphinx TOC entries (since I think the back button follows that hierarchy - it goes to the page that 'contains' the TOC listing of the current page).

**Comment by AndrewGardhouse** [comment]
> * The main demos page, the cards are all different heights. This is partially my fault - when I first made it, I wasn't sure how to get a 'grid' pattern, and so re-sized all the images so that the vertical card spacing was identical. But it would be nice to enforce this in the CSS, rather than by image + text height!

@josh146 I'm going to hold off on any more styling changes until we have finalized the copy and images for the cards on the main page

File: demonstrations.rst [hunk 1]
```diff
@@ -20,60 +20,114 @@ Demos
...
-                                <br>
-                                <strong>Community</strong>
-                                </h3>
-                                <p class="mb-1 grey-text px-3">Discover PennyLane demonstrations created by other users, or submit one yourself.</p>
-                                <div class="white-text d-flex justify-content-center"><h5>Read more <i class="fas fa-angle-double-right"></i></h5></div>
+                        <div class="card rounded-lg h-100">
+                            <div class="text-center d-flex align-items-center pb-2 h-100" style="justify-content: center;">
+                                <div class="d-flex flex-column h-100">
+                                    <h3 class="card-title">
+                                    <img src="_static/community.png" class="img-fluid" style="max-width: 130px;"></img>
```
Comment (josh146) [inline demonstrations.rst:121]: Might be good to reduce the size of this image, it looks quite large now compared to the others 😆

File: demonstrations.rst [hunk 2]
```diff
@@ -20,60 +20,114 @@ Demos
...
+            </div>
+
+            <div class="row justify-content-center main-cards">
+                <div class="col-lg-3 mb-4">
+                    <a href="demos_quantum-chemistry.html">
+                        <div class="card rounded-lg h-100">
+                            <div class="text-center d-flex align-items-center pb-2 h-100" style="justify-content: center;">
+                                <div class="d-flex flex-column h-100">
+                                    <h3 class="card-title">
+                                    <img src="_static/quantum-chemistry.png" class="img-fluid" style="max-width: 110px;"></img>
```
Comment (josh146) [inline demonstrations.rst:87]: Are the card heights still dependent on the value of `max-width` here?

File: demos_getting-started.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
+ .. role:: html(raw)
+   :format: html
+
+Getting Started
+======
```
Comment (josh146) [inline demos_getting-started.rst:5]:
```suggestion
Getting Started
===============
```

File: demos_getting-started.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.8.10/css/mdb.min.css" rel="stylesheet">
+
+:html:`<div class="gallery-grid row">`
+
+.. customgalleryitem::
+    :tooltip: Use quantum machine learning to rotate a qubit.
+    :figure: demonstrations/qubit_rotation/bloch.png
+    :description: :doc:`demos/tutorial_qubit_rotation`
+    :tags: autograd
```
Comment (josh146) [inline demos_getting-started.rst:23]: Slight reorganization here. Could we change the order as follows?

- Basic tutorial: qubit rotation
- Quantum gradients with backpropagation
- Plugins and hybrid computation
- Noisy circuits
- Gaussian transformation
- Computing gradients in parallel with Amazon Braket
- Using JAX with PennyLane
- Turning quantum nodes into Keras layers
- Turning quantum nodes into Torch layers
- Training a quantum circuit with PyTorch
- Pytorch and noisy devices
- Optimizing noisy circuits with cirq

This matches the google doc, but I think the Google doc must have been updated after you made the changes here, sorry!

File: demos_quantum-chemistry.rst [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    :tooltip: Learn about the variational quantum thermalizer algorithm, an extension of VQE.
+    :figure: demonstrations/vqt/thumbnail.png
+    :description: :doc:`demos/tutorial_vqt`
+    :tags: chemistry
+
+.. customgalleryitem::
+    :tooltip: VQE optimization using quantum natural gradient.
+    :figure: demonstrations/vqe_qng/vqe_qng_thumbnail.png
+    :description: :doc:`demos/tutorial_vqe_qng`
+    :tags: chemistry
```
Comment (josh146) [inline demos_quantum-chemistry.rst:62]: These ones I believe should be moved to the `Optimization` page, based on the latest google doc

File: demos_quantum-chemistry.rst [hunk 6]
```diff
@@ -0,0 +1,99 @@
+ .. role:: html(raw)
+   :format: html
+
+Quantum Chemistry
+============
```
Comment (josh146) [inline demos_quantum-chemistry.rst:5]:
```suggestion
Quantum Chemistry
=================
```

File: demos_quantum-chemistry.rst [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+.. meta::
+   :property="og:description": Implementations of the latest cutting-edge ideas and research from quantum machine learning using PennyLane.
+   :property="og:image": https://pennylane.ai/qml/_static/demos_card.png
```
Comment (josh146) [inline demos_quantum-chemistry.rst:9]: We should:

- Update the metadata description to be more specific to quantum chemistry
- Add an intro paragraph introducing quantum chemistry at the top of the page, as we do on the other demo categories.

@ixfoduap, any preferences for the content?

**Review by josh146 — COMMENTED** [review]
Thanks @AndrewGardhouse! new categories are looking really slick.

A couple of more general comments:

- @ixfoduap tweaked the demo ordering on the Google doc last week, which doesn't look like it is reflected here (sorry, I should have sent a message that it got updated!). So the orderings of the demo cards on each category page will need to be slightly modified.

- In addition a couple of demos have moved category - ones I could spot straight away include VQT and 'Accelerating VQE with QNG'.

- I'm curious if the `max-width` CSS rules on the category cards still affect the height/layout of the card grid?

- Finally, it might be time to update the page metadata (used for social media cards) to have distinct images per category page?

  They currently look like this (using the [Twitter card validator](https://cards-dev.twitter.com/validator)): 

  image

File: demonstrations.rst [hunk 8]
```diff
@@ -10,7 +10,7 @@ Demos
 =====
 
 .. meta::
-   :property="og:description": Take a deeper dive into quantum machine learning by exploring cutting-edge algorithms using PennyLane and near-term quantum hardware.
+   :property="og:description": Take a deeper dive into quantum computing by exploring cutting-edge algorithms using PennyLane and near-term quantum hardware.
```
Comment (ixfoduap) [inline demonstrations.rst:13]:
```suggestion
   :property="og:description": Take a deeper dive into quantum computing by exploring cutting-edge algorithms using PennyLane and quantum hardware.
```

File: demonstrations.rst [hunk 9]
```diff
@@ -20,60 +20,114 @@ Demos
     <div class="container">
         <!-- Section: Features v.1 -->
             <p class="lead grey-text text-center mx-auto mb-5">
-            Take a deeper dive into quantum machine learning by exploring cutting-edge algorithms using PennyLane and near-term quantum hardware.</p>
+            Take a deeper dive into quantum computing by exploring cutting-edge algorithms using PennyLane and near-term quantum hardware.</p>
```
Comment (ixfoduap) [inline demonstrations.rst:23]:
```suggestion
            Take a deeper dive into quantum computing by exploring cutting-edge algorithms using PennyLane and quantum hardware.</p>
```

File: demonstrations.rst [hunk 10]
```diff
@@ -20,60 +20,114 @@ Demos
...
+                    <a href="demos_quantum-chemistry.html">
+                        <div class="card rounded-lg h-100">
+                            <div class="text-center d-flex align-items-center pb-2 h-100" style="justify-content: center;">
+                                <div class="d-flex flex-column h-100">
+                                    <h3 class="card-title">
+                                    <img src="_static/quantum-chemistry.png" class="img-fluid" style="max-width: 110px;"></img>
+                                    <br>
+                                    <strong>Quantum Chemistry</strong>
+                                    </h3>
+                                    <p class="mb-1 grey-text px-3">Investigate, simulate, and infer properties of molecules and materials using quantum computing.</p>
```
Comment (ixfoduap) [inline demonstrations.rst:91]:
```suggestion
                                    <p class="mb-1 grey-text px-3">Study the properties of molecules and materials using quantum computing.</p>
```

File: demonstrations.rst [hunk 11]
```diff
@@ -20,60 +20,114 @@ Demos
...
+                    <a href="demos_quantum-computing.html">
+                        <div class="card rounded-lg h-100">
+                            <div class="text-center d-flex align-items-center pb-2 h-100" style="justify-content: center;">
+                                <div class="d-flex flex-column h-100">
+                                    <h3 class="card-title">
+                                    <img src="_static/quantum-computing.png" class="img-fluid" style="max-width: 66px;"></img>
+                                    <br>
+                                    <strong>Quantum Computing</strong>
+                                    </h3>
+                                    <p class="mb-1 grey-text px-3">Explore general quantum computing tasks and algorithms, from quantum volume to Boson sampling.</p>
```
Comment (ixfoduap) [inline demonstrations.rst:108]:
```suggestion
                                    <p class="mb-1 grey-text px-3">Explore general quantum computing tasks and algorithms, from quantum volume to boson sampling.</p>
```

File: demos_optimization.rst [hunk 12]
```diff
@@ -0,0 +1,154 @@
+ .. role:: html(raw)
+   :format: html
+
+Optimization
+============
+
+.. meta::
+   :property="og:description": Explore various topics and ideas, such as the shots-frugal Rosalin optimizer, the variational quantum thermalizer, or the issue of barren plateaus in quantum neural networks.
```
Comment (ixfoduap) [inline demos_optimization.rst:8]:
```suggestion
   :property="og:description": Explore various topics and ideas, such as the shots-frugal Rosalin optimizer, the variational quantum thermalizer, or barren plateaus in quantum neural networks.
```

File: demos_optimization.rst [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+   :property="og:image": https://pennylane.ai/qml/_static/demos_card.png
+
+Here you will find demonstrations showcasing quantum optimization. Explore
+various topics and ideas, such as the shots-frugal Rosalin
+optimizer, the variational quantum thermalizer, or the issue of barren
+plateaus in quantum neural networks.
```
Comment (ixfoduap) [inline demos_optimization.rst:14]:
```suggestion
optimizer, the variational quantum thermalizer, or barren plateaus
in quantum neural networks.
```

File: demos_optimization.rst [hunk 14]
```diff
@@ -0,0 +1,154 @@
...
+
+.. raw:: html
+
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.8.10/css/mdb.min.css" rel="stylesheet">
+
+:html:`<div class="gallery-grid row">`
+
+.. customgalleryitem::
+    :tooltip: Learn how to implement QAOA workflows with PennyLane
```
Comment (ixfoduap) [inline demos_optimization.rst:23]: Echoing what Josh mentioned, it would be great to reorder the demos in accordance to the google doc. My apologies for not communicating this change clearly enough. In this case the order should be

- Intro to QAOA
- Quantum natural gradient
- Accelerating VQEs with quantum natural gradient
- Barren plateaus in quantum neural networks
- Alleviating barren plateaus with local shot functions
- Optimization using SPSA
- Generalized parameter-shift rules
- Doubly stochastic gradient descent
- The Stochastic parameter-shift rule
- Quantum structure learning
- Frugal shot optimization with Rosalin
- Feedback-based quantum optimization (FALQON)
- Quantum analytic descent
- Variationally optimizing measurement protocols
- Variational quantum thermalizer
- Variational quantum linear solver
- Coherent quantum variational linear solver
- Optimizing a quantum optical neural network
- 3 qubit Ising model in Pytorch
- QAOA for maxcut

File: demos_quantum-chemistry.rst [hunk 15]
```diff
@@ -0,0 +1,99 @@
+ .. role:: html(raw)
+   :format: html
+
+Quantum Chemistry
+============
+
+.. meta::
+   :property="og:description": Implementations of the latest cutting-edge ideas and research from quantum machine learning using PennyLane.
```
Comment (ixfoduap) [inline demos_quantum-chemistry.rst:8]:
```suggestion
   :property="og:description": Explore and master quantum algorithms for quantum chemistry.
```

File: demos_quantum-chemistry.rst (hunk 7 above)
Comment (ixfoduap) [reply demos_quantum-chemistry.rst:9]: Intro paragraph:

Quantum chemistry is one the leading application areas of quantum computers. Master how to use PennyLane to construct molecular Hamiltonians, design quantum circuits for quantum chemistry, and simulate properties of molecules and materials.

File: demos_quantum-chemistry.rst [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+   :property="og:image": https://pennylane.ai/qml/_static/demos_card.png
+
```
Comment (ixfoduap) [inline demos_quantum-chemistry.rst:10]: Order here should be:

-Building molecular Hamiltonians
- A brief overview of VQE
- Givens rotations for quantum chemistry
- Modelling chemical reactions on a quantum computer
- Optimization of molecular geometries
- VQE in different spin sectors
- Measurement optimization
- VQE with parallel QPUs on Rigetti Forest

File: demos_quantum-computing.rst [hunk 17]
```diff
@@ -0,0 +1,75 @@
+ .. role:: html(raw)
+   :format: html
+
+Quantum Computing
+=================
+
+.. meta::
+   :property="og:description": Explore the applications of PennyLane to more general quantum computing tasks such as benchmarking and characterizing quantum processors.
```
Comment (ixfoduap) [inline demos_quantum-computing.rst:8]:
```suggestion
   :property="og:description": Explore the applications of PennyLane to general quantum computing tasks such as benchmarking and characterizing quantum processors.
```

File: demos_quantum-computing.rst [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+   :property="og:image": https://pennylane.ai/qml/_static/demos_card.png
+
+
+Explore the applications of PennyLane to more general quantum computing tasks
```
Comment (ixfoduap) [inline demos_quantum-computing.rst:12]:
```suggestion
Explore the applications of PennyLane to general quantum computing tasks
```

File: demos_quantum-computing.rst [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+such as benchmarking and characterizing quantum processors.
+
```
Comment (ixfoduap) [inline demos_quantum-computing.rst:14]: Order here:

- Quantum volume
- Understanding the Haar measure
- Classical shadows
- Quantum computation with neutral atoms
- Beyond classical computing with qsim
- Quantum advantage with Gaussian Boson Sampling

**Review by ixfoduap — CHANGES_REQUESTED** [review]
@AndrewGardhouse It looks so good! :star_struck: :star_struck: 

I left some suggestions on wording and proposed a header and intro text for the quantum chemistry section.

The main request is to please re-order the demos according to the google doc, to help us highlight key demos in each section. My apologies for not communicating this clearly enough from the start, I realize it's an annoying change to make :disappointed:

File: demonstrations.rst (hunk 2 above)
Comment (AndrewGardhouse) [reply demonstrations.rst:87]: Not the height of the cards, but the alignment of the content of the cards. The images aren't all the same size, so the `max-width` is being used to make the images look more consistent 🙂

File: demonstrations.rst (hunk 1 above)
Comment (AndrewGardhouse) [reply demonstrations.rst:121]: Does this work for you?
Screen Shot 2021-09-17 at 3 18 50 PM

**Comment by AndrewGardhouse** [comment]
> * Finally, it might be time to update the page metadata (used for social media cards) to have distinct images per category page?
>   They currently look like this (using the [Twitter card validator](https://cards-dev.twitter.com/validator)):
>   image

@josh146 I think this is out of scope for this ticket

File: demos_quantum-chemistry.rst (hunk 16 above)
Comment (AndrewGardhouse) [reply demos_quantum-chemistry.rst:10]: What about "The Variational Quantum Thermalizer" and "Adaptive circuits for quantum chemistry"?
Screen Shot 2021-09-17 at 4 55 08 PM

Comment (ixfoduap) [reply demos_quantum-chemistry.rst:10]: The variational thermalizer should be removed from here and used only in the optimization folder.

The adaptive circuits one is new, which is why we missed it. Great catch! It can go between Givens rotations and modelling reactions. So the order is

Building molecular Hamiltonians
A brief overview of VQE
Givens rotations for quantum chemistry
Adaptive circuits for quantum chemistry
Modelling chemical reactions on a quantum computer
Optimization of molecular geometries
VQE in different spin sectors
Measurement optimization
VQE with parallel QPUs on Rigetti Forest

**Comment by josh146** [comment]
> @josh146 I think this is out of scope for this ticket

Yes you're right 🙂

File: demonstrations.rst (hunk 2 above)
Comment (josh146) [reply demonstrations.rst:87]: Ah, got it!

File: demonstrations.rst (hunk 1 above)
Comment (josh146) [reply demonstrations.rst:121]: Looks great, thanks @AndrewGardhouse!

File: demos_optimization.rst (hunk 14 above)
Comment (AndrewGardhouse) [reply demos_optimization.rst:23]: @ixfoduap I can't find the demo `Alleviating barren plateaus with local shot functions` anywhere 🤔

Comment (josh146) [reply demos_optimization.rst:23]: @AndrewGardhouse it's this one https://pennylane.ai/qml/demos/tutorial_local_cost_functions.html 🙂

Comment (AndrewGardhouse) [reply demos_optimization.rst:23]: Thank you!

**Review by josh146 — APPROVED** [review]
Thanks @AndrewGardhouse, looks fantastic! Good to go from my end, but will wait for @ixfoduap's final review

**Review by ixfoduap — CHANGES_REQUESTED** [review]
Looks great! It just needs re-organizing the QML page according to the order in the google doc, which I replicated in the comments. Once that's done I'm happy to approve.

Thanks @AndrewGardhouse for the amazing work! :man_dancing: :tada:

File: demos_qml.rst [hunk 20]
```diff
@@ -0,0 +1,113 @@
+ .. role:: html(raw)
+   :format: html
+
+Quantum machine learning
+========================
+
```
[the remark this answers is not in the document]
Comment (AndrewGardhouse) [reply demos_qml.rst:6]: @ixfoduap Done 🙂

**Review by ixfoduap — APPROVED** [review]
:shipit:

**Comment by co9olguy** [comment]
:tada:
